src/components/Geolocation/Geolocation.js

```diff
@@ -127,10 +130,17 @@ class Geolocation extends PureComponent {
       'EPSG:4326',
       code,
     );
-
-    const point = new Point(pos);
-    this.highlight(point);
-    this.layer.setMap(map);
+    if (!point) {
+      var tmpPoint = new Point(pos);
+      this.setState({
+        point: tmpPoint,
+      });
+      this.highlight(tmpPoint);
+      this.layer.setMap(map);
+    } else {
+      point.setCoordinates(pos);
+    }
```

Comment:
12 lines of code instead of 3  only to avoid the creation of a point once in a while, it seems a bit overkill. Optimisation is good only when you gain something (performance or code readibility).   And  putting the point in the state is totally useless  since you never update it, so you trigger a useless render of the component.

Reply:
I see where the problem was . I thought activate was the function call when you click on the button which is not . So this make sense but the setState is still useless.

Reply:
Yes, the function gets called for every location update which can be very often. This change makes a huge performace difference on my smartphone

> but the setState is still useless.
I need to store the point somewhere to update the position. Can I do that without the setState?

Reply:
I will try to refactor it that way